src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/PayRatesService.kt

```diff
@@ -63,116 +63,97 @@ class PayRatesService(
       }
 
   /*
-   * Rebuild the list of pay rates to replace the existing list - taking into account we never delete old rates, just expire them
+   * Rebuild the list of pay rates to replace the existing list.
    *
-   * This is quite a tricky algorithm. When building the new rate list it does the following:
-   * - newly requested rates are created and become effective today
-   * - any existing rates that are still active but haven't changed are retained
-   * - any existing rates that have changed are expired with today's date and a new rate added effective tomorrow
-   * - any existing rates that are expired are retained
-   * - any existing rates that are not included in the new request are expired
+   * This is complicated by the following:
+   * - DPS doesn't support end dates so we have to derive them from the start dates
+   * - DPS allows hard deletes of rates, but we can't delete them from NOMIS
+   *
+   * To handle the delete problem we expire rates in NOMIS that are no longer in the DPS requested rates.
    */
-  fun buildNewPayRates(requestedPayRates: List<PayRateRequest>, existingActivity: CourseActivity, previousActivityEndDate: LocalDate?): MutableList<CourseActivityPayRate> {
-    val newPayRates = mutableListOf<CourseActivityPayRate>()
-    val existingPayRates = existingActivity.payRates
-    val tomorrow = LocalDate.now().plusDays(1)
-
-    requestedPayRates.forEach { requestedPayRate ->
-      val existingPayRate = existingPayRates.findExistingPayRate(requestedPayRate)
+  fun buildNewPayRates(requestedPayRates: List<PayRateRequest>, existingActivity: CourseActivity): List<CourseActivityPayRate> {
```

Comment:
The main change is to rewrite the algorithm to handle the new DPS model. 

It would be great if their model matched NOMIS then we could perform a straight mapping. Unfortunately deleting pay rate history from NOMIS is not an option because people can be paid in the future so we need to keep the old rates.